detection_rules/rule.py

```diff
@@ -406,15 +406,17 @@ def lock_info(self, bump=True) -> dict:
     @property
     def is_dirty(self) -> Optional[bool]:
         """Determine if the rule has changed since its version was locked."""
-        existing_sha256 = self.version_lock.get_locked_hash(self.id, self.metadata.get('min_stack_version'))
+        min_stack = self.metadata.get('min_stack_version') or str(get_min_supported_stack_version(drop_patch=True))
```

Comment:
add this `or str(get_min_supported_stack_version` here and in L418 is the primary fix to this bug